pkg/rlptools/wasm_utils.go

```diff
@@ -10,98 +10,100 @@ import (
 	istioclientgoextensionv1alpha1 "istio.io/client-go/pkg/apis/extensions/v1alpha1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	gatewayapiv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+	gatewayapiv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
 
-	kuadrantv1beta1 "github.com/kuadrant/kuadrant-operator/api/v1beta1"
+	kuadrantv1beta2 "github.com/kuadrant/kuadrant-operator/api/v1beta2"
 	"github.com/kuadrant/kuadrant-operator/pkg/common"
 )
 
 var (
 	WASMFilterImageURL = common.FetchEnv("RELATED_IMAGE_WASMSHIM", "oci://quay.io/kuadrant/wasm-shim:latest")
+
+	PathMatchTypeMap = map[gatewayapiv1alpha2.PathMatchType]PatternOperator{
+		gatewayapiv1beta1.PathMatchExact:             PatternOperator(kuadrantv1beta2.EqualOperator),
+		gatewayapiv1beta1.PathMatchPathPrefix:        PatternOperator(kuadrantv1beta2.StartsWithOperator),
+		gatewayapiv1beta1.PathMatchRegularExpression: PatternOperator(kuadrantv1beta2.MatchesOperator),
+	}
 )
 
-type GatewayAction struct {
-	Configurations []kuadrantv1beta1.Configuration `json:"configurations"`
+type SelectorSpec struct {
+	// Selector of an attribute from the contextual properties provided by Envoy
+	// during request and connection processing
+	// https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/advanced/attributes
+	// They are named by a dot-separated path (e.g. request.path)
+	// Examples:
+	// "request.path" -> The path portion of the URL
+	Selector string `json:"selector"`
```

Comment:
What about defining a `type WellknownSelector string`?

```suggestion
	Selector WellknownSelector `json:"selector"`
```

It would also serve as the type for `PatternExpression.Selector`:

https://github.com/Kuadrant/kuadrant-operator/blob/5c062052bad1cde084855129cd9429dd3fb957ad/pkg/rlptools/wasm_utils.go#L72

Reply:
in https://github.com/Kuadrant/kuadrant-operator/blob/3966aa6b02f711af7dd44a7bfc6091f3a3ecc51b/api/v1beta2/ratelimitpolicy_types.go#L39 we use `ContextSelector`. Shall we change that as well?

Reply:
using `ContextSelector` in WASM plugin Configuration object